subliminal/providers/podnapisi.py

```diff
@@ -74,57 +67,56 @@ def __init__(self):
-            params['sTE'] = episode
+            params['seasons'] = season
+            params['episodes'] = episode
+            params['movie_type'] = ['tv-series', 'mini-series']
+        else:
+            params['movie_type'] = 'movie'
         if year:
-            params['sY'] = year
+            params['year'] = year
 
         # loop over paginated results
         logger.info('Searching subtitles %r', params)
         subtitles = []
         pids = set()
         while True:
             # query the server
-            r = self.session.get(self.server_url + 'search/old', params=params, timeout=10)
+            r = self.session.get(self.server_url + 'search/advanced', params=params, timeout=10)
             r.raise_for_status()
-            xml = etree.fromstring(r.content)
-
-            # exit if no results
-            if not int(xml.find('pagination/results').text):
-                logger.debug('No subtitles found')
-                break
+            result = json.loads(r.text)
 
             # loop over subtitles
-            for subtitle_xml in xml.findall('subtitle'):
+            for data in result['data']:
                 # read xml elements
-                pid = subtitle_xml.find('pid').text
+                pid = data['id']
                 # ignore duplicates, see http://www.podnapisi.net/forum/viewtopic.php?f=62&t=26164&start=10#p213321
                 if pid in pids:
+                    logger.debug('Ignoring duplicate %r', pid)
+                    continue
+
+                if is_episode and data['movie']['type'] == 'movie':
+                    logger.error('Wrong type detected: movie for episode')
                     continue
 
-                language = Language.fromietf(subtitle_xml.find('language').text)
-                hearing_impaired = 'n' in (subtitle_xml.find('flags').text or '')
-                page_link = subtitle_xml.find('url').text
-                releases = []
-                if subtitle_xml.find('release').text:
-                    for release in subtitle_xml.find('release').text.split():
-                        release = re.sub(r'\.+$', '', release)  # remove trailing dots
-                        release = ''.join(filter(lambda x: ord(x) < 128, release))  # remove non-ascii characters
-                        releases.append(release)
-                title = subtitle_xml.find('title').text
-                season = int(subtitle_xml.find('tvSeason').text)
-                episode = int(subtitle_xml.find('tvEpisode').text)
-                year = int(subtitle_xml.find('year').text)
+                language = Language.fromietf(data['language'])
+                hearing_impaired = 'hearing_impaired' in data['flags']
+                page_link = data['url']
+                releases = data['releases'] + data['custom_releases']
+                title = data['movie']['title']
+                season = int(data['movie']['episode_info']['season']) if is_episode else None
```

Comment:
Season and episode may not be defined. I believe using a `get` with apropriate default value would be a better choice.